### ISSUE TITLE
Automated deployments of the kit example app (i.e. vercel, netlify, cloudflare) fail with 50x

### ISSUE THREAD

**Issue by spacejunky** [sveltejs/kit#4135]
### Describe the bug

Looking for guidance on how to deploy to Cloudflare. Tried the 'official' example deployment from "kit/packages/create-svelte/templates/default".

All 3 targets produce one or another 50x code on execution and fail correct function when manipulating TODO's. Console Logs below, server logs not accessible to me.

- cloudflare 500's when attempting to modify the done state of a todo
- netlify 502's on opening the .../todos route 
- vercel 500's on attempting to add a todo

### Reproduction

No special setup required. Simply browse to one of the links below, open browser console and attempt to create/manipulate todo's.

- https://cloudflare.demo.svelte.dev/
- https://kit-demo.netlify.app/
- https://kit-demo.vercel.app/

### Logs

```shell
Cloudflare occurence
====================
VM12:1          POST https://cloudflare.demo.svelte.dev/todos?_method=PATCH 500
(anonymous) @ VM12:1
y @ index.svelte-ec10a7c1.js:1

index.svelte-ec10a7c1.js:1 <!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8" />
		<meta name="description" content="Svelte demo app" />
		<link rel="icon" href="./favicon.png" />
		<meta name="viewport" content="width=device-width, initial-scale=1" />
		
	<link rel="stylesheet" href="/_app/assets/pages/__layout.svelte-a7861b78.css">
	<link rel="modulepreload" href="/_app/start-99527e01.js">
	<link rel="modulepreload" href="/_app/chunks/vendor-062b0a54.js">
	<link rel="modulepreload" href="/_app/chunks/singletons-a6a7384f.js">
	<link rel="modulepreload" href="/_app/pages/__layout.svelte-df613146.js">
	<link rel="modulepreload" href="/_app/error.svelte-f6b7cd73.js">
	</head>
	<body>
		<div>


<header class="svelte-t2wq17"><div class="corner svelte-t2wq17"><a href="https://kit.svelte.dev" class="svelte-t2wq17"><img src="/_app/assets/svelte-logo-87df40b8.svg" alt="SvelteKit" class="svelte-t2wq17"></a></div>

	<nav class="svelte-t2wq17"><svg viewBox="0 0 2 3" aria-hidden="true" class="svelte-t2wq17"><path d="M0,0 L1,2 C1.5,3 1.5,3 2,3 L2,0 Z" class="svelte-t2wq17"></path></svg>
		<ul class="svelte-t2wq17"><li class="svelte-t2wq17"><a sveltekit:prefetch href="/" class="svelte-t2wq17">Home</a></li>
			<li class="svelte-t2wq17"><a sveltekit:prefetch href="/about" class="svelte-t2wq17">About</a></li>
			<li class="svelte-t2wq17 active"><a sveltekit:prefetch href="/todos" class="svelte-t2wq17">Todos</a></li></ul>
		<svg viewBox="0 0 2 3" aria-hidden="true" class="svelte-t2wq17"><path d="M0,0 L0,3 C0.5,3 0.5,3 1,2 L2,0 Z" class="svelte-t2wq17"></path></svg></nav>

	<div class="corner svelte-t2wq17"></div>
</header>

<main class="svelte-1izrdc8"><h1>500</h1>

<pre>Illegal invocation</pre>




<pre>TypeError: Illegal invocation</pre></main>

<footer class="svelte-1izrdc8"><p>visit <a href="https://kit.svelte.dev" class="svelte-1izrdc8">kit.svelte.dev</a> to learn SvelteKit</p>
</footer>


		<script type="module" data-hydrate="1in42mn">
		import { start } from "/_app/start-99527e01.js";
		start({
			target: document.querySelector('[data-hydrate="1in42mn"]').parentNode,
			paths: {"base":"","assets":""},
			session: {},
			route: true,
			spa: false,
			trailing_slash: "never",
			hydrate: {
				status: 500,
				error: {name:"TypeError",message:"Illegal invocation",stack:"TypeError: Illegal invocation"},
				nodes: [
					import("/_app/pages/__layout.svelte-df613146.js"),
						import("/_app/error.svelte-f6b7cd73.js")
				],
				params: {}
			}
		});
	</script></div>
	</body>
</html>

y @ index.svelte-ec10a7c1.js:1


Netlify occurrence
==================

VM9:1          GET https://kit-demo.netlify.app/todos/__data.json 502
(anonymous) @ VM9:1
_load @ start-551abaaa.js:1
await in _load (async)
_get_navigation_result @ start-551abaaa.js:1
load @ start-551abaaa.js:1
prefetch @ start-551abaaa.js:1
e @ start-551abaaa.js:1
(anonymous) @ start-551abaaa.js:1
setTimeout (async)
r @ start-551abaaa.js:1


Vercel occurrence
=================

VM9:1          POST https://kit-demo.vercel.app/todos.json 500
(anonymous) @ VM9:1
s @ index.svelte-120584f4.js:1
VM9:1          POST https://kit-demo.vercel.app/todos.json 500
(anonymous) @ VM9:1
s @ index.svelte-120584f4.js:1
index.svelte-120584f4.js:1 Cannot read property 'get' of undefined
s @ index.svelte-120584f4.js:1
```

### System Info

```shell
* My svelte environment not relevant as builds were presumably part of svelte CI

* My browser, M1 Macbook Air:
    Chrome, Version 98.0.4758.109 (Official Build) (arm64)
```

### Severity

annoyance

### Additional Information

Preventing me deploying to any of these hosting services (for me Cloudflare is preferable choice, planning to eventually use durable objects).

Three superficially different failures submitted as one Issue as I assume there must be a common factor for all 3 to fail simultaneously; sadly my lack of knowledge of Svelte and/or the hosters does not allow me to discern this cause ;-(

**Comment by manuganji** [comment]
I can create the todos. You are referring to your own deployment right?

**Comment by spacejunky** [comment]
No, not mine.

These are deployments that I understand to be automatically done as part of the CI process. At least the text in the README I refer to says:

```
The default template is automatically deployed to Cloudflare Pages, Netlify and Vercel on every commit. Commits to master will update production deployments:

https://cloudflare.demo.svelte.dev/
https://kit-demo.netlify.app/
https://kit-demo.vercel.app/

```

Creating TODO's is possible (in the cloudfare deploy), the problem there appears when you try to mark them as done ...

**Comment by manuganji** [comment]
I think this issue originated in this change: https://github.com/sveltejs/kit/commit/a37f3b3de07304641cdae5d3686bfda6861b4489

**Comment by geoffrich** [comment]
I see slightly different behavior from the original issue report:

- I get a 500 when attempting to add a TODO on the [automated Vercel deployment](https://kit-demo.vercel.app/todos). However, I do not get any errors on a [fresh deploy](https://sveltekit-demo-lime.vercel.app/todos) of the demo app to Vercel. Since it works on a manual deployment, the automated deployment might be out of date.
- On both Netlify and Cloudflare, I get a 5xx when attempting to modify a TODO. Netlify responds with "_method=PATCH is only allowed with POST requests" and Cloudflare responds with "TypeError: Illegal invocation". This is true both with the automated deployments, as well as demo apps I deployed myself.

**Comment by spacejunky** [comment]
I can confirm that my own deployment to Vercel works 100%. And this is the smoothest experience of the three, by a long way.

I have still failed to get either Cloudflare or Netlify to work correctly and to be frank I've now given up on trying those two and centered on Vercel as I need to progress on my project.

**Comment by janat08** [comment]
Can I work on this for a bounty?

**Comment by jhubbardsf** [comment]
Went to look at this bug. Created a demo app and deployed it to netlify (can see it [here](https://todo-bug-test.netlify.app/todos)). Created Todos works. Trying to delete a Todo results in 502. 

`index.svelte-b7508773.js:1 {"errorType":"Error","errorMessage":"_method=DELETE is only allowed with POST requests","trace":["Error: _method=DELETE is only allowed with POST requests","    at respond (/var/task/.netlify/server/index.js:2112:13)","    at Server.respond (/var/task/.netlify/server/index.js:2383:12)","    at Runtime.handler (/var/task/.netlify/serverless.js:53:33)","    at Runtime.handleOnce (/var/runtime/Runtime.js:66:25)"]}`

Same while trying to edit a Todo:

`index.svelte-b7508773.js:1 {"errorType":"Error","errorMessage":"_method=PATCH is only allowed with POST requests","trace":["Error: _method=PATCH is only allowed with POST requests","    at respond (/var/task/.netlify/server/index.js:2112:13)","    at Server.respond (/var/task/.netlify/server/index.js:2383:12)","    at Runtime.handler (/var/task/.netlify/serverless.js:53:33)","    at Runtime.handleOnce (/var/runtime/Runtime.js:66:25)"]}`

Usually I can reproduce Netlify errors locally by using the Netlify adapter and running the app with preview, however I can't reproduce it locally at all. Just adding the info here.

**Comment by Conduitry** [comment]
Interesting. The redirect after the delete is a 303 to `/todos?_method=DELETE` even though it looks like the query param shouldn't be there: https://github.com/sveltejs/kit/blob/0cabe776a04c250513dee2514dc7c4c298a702ba/packages/create-svelte/templates/default/src/routes/todos/index.ts#L48 I wonder whether Netlify is doing something weird to transform the response.

**Comment by jhubbardsf** [comment]
Yeah, I think this is all related to the methodOverrides.

```ts
	kit: {
		adapter: adapter(),

		// Override http methods in the Todo forms
		methodOverride: {
			allowed: ['PATCH', 'DELETE']
		}
	}
```

**Comment by FunMiles** [comment]
On Vercel deployments I get the following errors. They seem related to the lines @jhubbardsf noted:

> Unexpected option config.kit.methodOverride
--
11:11:58.698 | Error: Unexpected option config.kit.methodOverride
11:11:58.698 | at validate (file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:380:10)
11:11:58.699 | at validate (file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:399:19)
11:11:58.699 | at validate_config (file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:488:20)
11:11:58.699 | at load_config (file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:465:20)
11:11:58.699 | at async get_config (file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:519:10)
11:11:58.699 | at async file:///vercel/path0/node_modules/@sveltejs/kit/dist/cli.js:604:18
11:11:58.721 | Error! Command "npm run build" exited with 1
11:11:58.821 | Error: Command "vercel build" exited with 1

**Comment by mefaba** [comment]
I believe links in the reproduction section of the issue post are outdated.  @spacejunky is this bug still exist?

**Comment by spacejunky** [comment]
@mefaba there are two answers to this question, I think ...

*1)* _NO_, because a) the example app has been updated, removing the part which used to cause the problems and b) the automated deployments, at least to netlify & vecel, seem not to be done any more. The surface issue of the example app failing is therefore overtaken by events and no longer relevant.

*2)* _I DON'T KNOW_ if the underlying bug is still there or not. As of June/July it seems that it was still an issue, potentially related to method overrides.

I have worked around the whole thing by only using vercel and avoiding method overrides. Searching for the method override issue exceeds my Sveltekit knowledge, and available time, just now.
